src/components/StatusBar/StatusBarToggleButton.tsx

```diff
@@ -0,0 +1,60 @@
+import {
+    IconButton,
+    IconButtonProps,
+    Tooltip,
+    TooltipTypeMap,
+} from "@mui/joy";
+
+import CheckIcon from "@mui/icons-material/Check";
+import CloseIcon from "@mui/icons-material/Close";
+
+
+interface MenuBarToggleButtonProps extends IconButtonProps {
+    tooltipPlacement?: TooltipTypeMap["props"]["placement"];
+    tooltipTitle?: string;
+    isActive?: boolean;
+    onIcon?: React.ReactNode;
+    offIcon?: React.ReactNode;
```

Comment:
how about
```suggestion
    isActive: boolean;
    icons?: {
        active?: React.ReactNode;
        inactive?: React.ReactNode;
    };
    tooltipPlacement?: TooltipTypeMap["props"]["placement"];
    tooltipTitle?: string;
```
1. Let's require `isActive` since this is a toggle button.
2. I proposed a different interface for setting the icons.
3. the order of the properties are alphabetized.